### PR TITLE
feat(analyzer-embed): scaffold crate (chunking + sidecar + CLI)

### PR DESCRIPTION
**Pull request by avifenesh**
First slice of #27. Lands the data plumbing so subsequent PRs can plug in the ONNX model layer behind a stable trait boundary.

## What's in this PR

- New crate `analyzer-embed` produces binary `agent-analyzer-embed`
- **`chunk` module** — per-file or per-function chunking via tree-sitter for Rust / TS / JS / Python / Go / Java; markdown chunked by heading; falls back to per-file for unsupported languages
- **`sidecar` module** — packed fp16 binary format for vector storage (2 bytes per dim, deterministic ordering across runs, fast load)
- **`schema` module** — JSON document format piped to `agent-analyzer set-embeddings` (mirrors the existing `set-descriptors` pattern)
- **`embedder` trait** — stable API surface; concrete ONNX implementation ships in the next PR
- CLI subcommands `scan` and `update` parse arguments and exit with a clear `NOT IMPLEMENTED` message rather than emitting fake JSON
- 18 unit tests covering chunking edge cases (per-file vs per-function, empty/whitespace, unsupported extensions, struct-as-Type, line numbering, markdown sections) and sidecar round-trips (empty, populated, deterministic write, fp16 precision tolerance, length-mismatch error)

## What's deferred to next PR

- ONNX Runtime (`ort` crate) integration
- HuggingFace `tokenizers` integration
- Model file download + caching next to the binary
- Two model variants: BGE-small-en-v1.5 Q8 (small) and EmbeddingGemma-300M Q4 (big)
- Actual `scan` and `update` execution paths
- `agent-analyzer set-embeddings` subcommand on the main binary

## Why scaffold-first

The chunking strategy and on-disk format are the parts most likely to need revision based on real use; getting them in front of CI and reviewers before adding the heavy ONNX dependency keeps each PR focused. Subsequent PRs can land the model integration without re-litigating these design choices.

## Test results

\`\`\`
running 18 tests
test result: ok. 18 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
\`\`\`

Workspace builds clean (`cargo check --workspace`).

## Refs

- #27 — full design
- agent-sh/repo-intel#16 — companion skill orchestration

## Test Plan

- [x] `cargo test -p analyzer-embed` passes
- [x] `cargo check --workspace` clean
- [ ] Reviewers verify trait boundary makes sense for the planned ONNX implementation
- [ ] CI green

---

> [!NOTE]
> **Medium Risk**
> Adds a new workspace crate and binary plus a new fp16 sidecar file format; while not wired into the main CLI yet, the parsing/serialization logic and on-disk format could impact future compatibility and correctness.
> 
> **Overview**
> Introduces a new workspace crate, `analyzer-embed`, and a new `agent-analyzer-embed` CLI intended to generate and update local repo embeddings (currently argument parsing only, with `scan`/`update` returning a clear not-implemented exit).
> 
> Implements the core data plumbing: tree-sitter/markdown-based source chunking (`chunk_file`), a JSON interchange schema (`EmbeddingsDocument`) for piping results to a future `set-embeddings` command, and a deterministic fp16-packed binary sidecar format (`Sidecar`) for storing vectors. Workspace dependencies/lockfile are updated to include the new crate and its crypto/fp16-related deps.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit d79feded61fd005e2911b06dc1e6a850e2dc9551. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: crates/analyzer-embed/src/chunk.rs [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+        }
+        if let Some(node) = decl_node {
+            let start = node.start_position();
+            let end = node.end_position();
+            let text = content[node.byte_range()].to_string();
+            let kind = if name_text.as_deref().map(is_type_name).unwrap_or(false) {
+                ChunkKind::Type
+            } else {
+                ChunkKind::Function
+            };
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/analyzer-embed/src/chunk.rs]: Fixed in 55d542b — replaced is_type_name name-capitalization heuristic with chunk_kind_for_node(node.kind()) using the actual AST node kind. Added test go_exported_function_is_classified_as_function_not_type that exercises HandleRequest/helper/MyStruct.

File: crates/analyzer-embed/src/chunk.rs [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+  (class_declaration name: (type_identifier) @name) @decl
+  (interface_declaration name: (type_identifier) @name) @decl
+  (export_statement (function_declaration name: (identifier) @name)) @decl
+  (export_statement (class_declaration name: (type_identifier) @name)) @decl
+  (lexical_declaration
+    (variable_declarator
+      name: (identifier) @name
+      value: [(arrow_function) (function_expression)])) @decl
+]
+"#;
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/analyzer-embed/src/chunk.rs:181]: Fixed in 55d542b — removed (export_statement (function_declaration …)) and (export_statement (class_declaration …)) wrapped patterns from both TS_QUERY and JS_QUERY. The bare patterns already match the inner declaration regardless of wrapping. Added test ts_exported_function_yields_single_chunk_not_two.
